### PR TITLE
Merge from official twilio-flutter/programmable-video repo

### PR DESCRIPTION
**Pull request by samfong-happyco**
Merge from the official repo to support null safety and bug fixes.

Retain the following custom changes:
- Take photo functionality
- 720p 24fps capture resolution

Added new functionality:
- Hit test behaviour option for the native widget

### CODE REVIEW

File: programmable_video/android/src/main/kotlin/twilio/flutter/twilio_programmable_video/VideoFrameKtx.kt [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            i420Buffer,
+            buffer.width,
+            buffer.height
+    )
+    val stream = ByteArrayOutputStream()
+    val rect =
+            Rect(0, 0, yuvImage.width, yuvImage.height)
+
+    // Compress YuvImage to jpeg
+    yuvImage.compressToJpeg(rect, 100, stream)
```
Comment (adammoyle) [inline programmable_video/android/src/main/kotlin/twilio/flutter/twilio_programmable_video/VideoFrameKtx.kt:40]: 100 seems high, what is the file size for an image with 100?

File: programmable_video/android/src/main/kotlin/twilio/flutter/twilio_programmable_video/VideoFrameKtx.kt [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+            bitmap!!, 0, 0, bitmap.width, bitmap.height, matrix, true
+    )
+    return bitmap
+}
+
+fun VideoFrame.toJpeg(): ByteArray? {
+    val bitmap = toBitmap()
+    return if (bitmap != null) {
+        val stream = ByteArrayOutputStream()
+        bitmap.compress(Bitmap.CompressFormat.JPEG, 100, stream)
```
Comment (adammoyle) [inline programmable_video/android/src/main/kotlin/twilio/flutter/twilio_programmable_video/VideoFrameKtx.kt:71]: Same as above

File: programmable_video/lib/src/camera_capturer.dart [hunk 3]
```diff
@@ -89,10 +89,9 @@ class CameraCapturer implements VideoCapturer {
   }
 
   /// Takes a photo from the camera capturer.
-  Future<dynamic> takePhoto(int imageCompression) async {
```
Comment (adammoyle) [inline programmable_video/lib/src/camera_capturer.dart:93]: Why remove this?

File: programmable_video/android/src/main/kotlin/twilio/flutter/twilio_programmable_video/VideoFrameKtx.kt (hunk 1 above)
Comment (samfong-happyco) [reply programmable_video/android/src/main/kotlin/twilio/flutter/twilio_programmable_video/VideoFrameKtx.kt:40]: Roughly 740 kb.

File: programmable_video/android/src/main/kotlin/twilio/flutter/twilio_programmable_video/VideoFrameKtx.kt (hunk 2 above)
Comment (samfong-happyco) [reply programmable_video/android/src/main/kotlin/twilio/flutter/twilio_programmable_video/VideoFrameKtx.kt:71]: Actually, its roughly 740 kb for iOS and about 500 kb for Android.

File: programmable_video/lib/src/camera_capturer.dart (hunk 3 above)
Comment (samfong-happyco) [reply programmable_video/lib/src/camera_capturer.dart:93]: It was harder to pass through the imageCompression option to the new Android implementation. Also, we're not really use of this option in the first place, as it was defaulted to 80 before.

**Comment by adammoyle** [comment]
LGTM

**Pull request merged**